lego/apps/users/models.py

```diff
@@ -552,24 +552,33 @@ class PhotoConsent(BasisModel):
     class Meta:
         unique_together = ("semester", "year", "domain", "user")
 
-    def get_consents(self, user):
+    def get_consents(self, user, *, time=None):
         now = timezone.now()
-        current_semester = PhotoConsent.get_semester(now)
-        current_year = now.year
-
-        PhotoConsent.objects.get_or_create(
-            user=user,
-            year=current_year,
-            semester=current_semester,
-            domain=constants.SOCIAL_MEDIA_DOMAIN,
-        )
+        consent_time = now
+        if time is not None:
+            consent_time = time
+        consent_semester = PhotoConsent.get_semester(consent_time)
+        consent_year = consent_time.year
+
+        # Don't create PhotoConsent objects for the past
+        if consent_time >= now:
+            PhotoConsent.objects.get_or_create(
```

Comment:
Could it be an idea to loop over the domains in the constants in case we add more domains in the future?

Reply:
Yes!
I thought about it, but wanted this PR to just cover the bug it's fixing.
I could fix up all of the consent code as well, but it will make the PR larger as well as make more time.


Premature optimization is a thing, and there is no point in refactoring code before it's necessary. I think we should leave this until we _actually_ add another domain, then we can refactor and fix this code. Otherwise you just get caught in an endless refactoring and code change loop.

To quite Martin Fowler from the book _refactoring_[^1]:
> The first time you do something, you just do it. The second time you do something similar, you wince at the duplication, but you do the duplicate thing anyway. The third time you do something similar, you refactor.

[^1]: https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)

Reply:
Okay, fair enough